### ISSUE TITLE
Playing the same sound multiple times while it is still playing

### ISSUE THREAD

**Issue by ibiza240** [leoasis/react-sound#14]
Hi, great component indeed, thanks to you!

I was wondering if it's possible to play the sound multiple times, let's say if I click on my button playing the sound multiple times in quick succession?

From the soundmanager2 demo site, http://www.schillmania.com/projects/soundmanager2/demo/api/
it seems to be the default behaviour?

However from my app, the sound won't play again until it has completed playing.

Any idea how to achieve this? Thanks! :)

**Comment by leoasis** [comment]
Hey! Sorry for the late response.

The idea of the library is that you can treat each sound as a component. To play multiple sounds, just render multiple Sound components! That should work

**Comment by ibiza240** [comment]
Hey! Thanks for the response :)

Here is my simple scenario : a sound should played each time the user clicks the button. I want it to play from the start even if the sound from the previous click is stil playing (and let it finish, do not cut the previous sound). Should I render like 10 sound components to be sure that a new sound will play each time?

**Comment by leoasis** [comment]
Hey! Sorry for taking to long to answer, was out on vacations.

Regarding what you want to do, you could add a sound description object into your local component state every time you click the button, and render as many `Sound` components as there are in the list of sounds in the local state. The sound description object is just an object that just has the necessary information to render the sound later. The idea is something like this:

``` js
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = { sounds: [] };
  }

  handleClick() {
    this.setState({ sounds: this.state.sounds.concat({ url: 'some-url.mp3' }) });
  }

  render() {
    return (
      <div>
        <button onClick={this.handleClick.bind(this)}>Click me</button>
        {this.state.sounds.map((sound) => <Sound url={sound.url} {...whatever} />)}
      </div>
    );
  }
}
```

**Comment by leoasis** [comment]
Hope the answer above helped you! If not, feel free to reopen this issue, closing it since it's a little dated

**Comment by leizard** [comment]
hey @leoasis. I implemented as you recommended. Everything works find except one issue: sometimes when I click the button, it plays 2 sound at the same time. Seem like the old Sound component is not destroyed and it still play.
Do you have any idea on this?
Thank you.

**Comment by leoasis** [comment]
That could possible be a bug, but to understand better what's going on, can you provide an example or a repo with the minimal setup that reproduces the issue you're seeing?

**Comment by ELI7VH** [comment]
Hello. Thank you for this! Makes way more sense now. 

I am also experiencing the same bug. here is my code, and here is a development build of my project: 209.89.72.56.8080 (sorry I can't figure out how to put this onto codepen right now (just woke up). If i figure out a codepen, I will update the link. 

https://pastebin.com/DrJPnwUw

secondary question, but is there a way to reduce the latency of the click and sound?

**Comment by leoasis** [comment]
@elijahlucian according to that code you are playing all the sounds in the array. Do you really need multiple sounds? If so, do you want them to play at the same time? You should control the `playStatus` of the sounds in the state to achieve that.

**Comment by ELI7VH** [comment]
I just copied the code as you wrote above. Like the guy above you said, we just want a single sound to play every time the gui is pressed, regardless of if the current sound is done playing or not. 

I fixed the link to my project : http://209.89.72.56:3000 

take a look.  - so we just want a sound to play when the button is pressed. but sometimes it plays 2 sounds. sometimes it starts building up.

**Comment by leoasis** [comment]
If you want to only play one sound at a time, then you can:

1. render a single `Sound` component. That would still need to load the sound before it starts playing.
2. Render multiple sounds and just have one with `playStatus={Sound.playStatus.PLAYING}` and moving forward.
